scribble-lib/scribble/base.rkt

```diff
@@ -325,33 +309,30 @@
 ;; ----------------------------------------
 
 (define (cell-spec/c c)
-  (define rc
-    (recursive-contract (or/c c
-                              empty
-                              (cons/c rc rc))))
-  rc)
+  (recursive-contract (or/c c empty (cons/c rc rc))))
```

Comment:
Whoa!

Reply:
@jackfirth this needs a side condition to make sure that `rc` doesn't occur in the body of `define`.

Reply:
If the inlined definition binds a (non-recursive) lambda expression, would the name be lost?

Reply:
Yes, but that doesn't happen because first those get rewritten into function definitions of the form `(define (f) ...)` (as opposed to `(define f (lambda () ...))`). Then, there's a check in the inlining rule to not inline function definitions, only variable definitions.

Reply:
Wow, this is very cool!